### ISSUE TITLE
VS 2017 support

### ISSUE THREAD

**Issue by kayru** [deplinenoise/tundra#289]
It would be great to have VS 2017 IDE project generator and toolchain support.

**Comment by deplinenoise** [comment]
Agree! You volunteering? :)

**Comment by kayru** [comment]
If it's as simple as VS 2015 support, then sure :)

**Comment by kayru** [comment]
FYI, I have a super hacky support for vs2017 here: https://github.com/kayru/tundra/commits/vs2017
Though I really have no idea what I'm doing or how to do it properly :)

**Comment by deplinenoise** [comment]
That looks about right. If it works for you, and doesn't break older
versions, can you rebase and send a PR?

Thanks!

Sent from my iPhone

On Sep 10, 2017, at 03:30, kayru <notifications@github.com> wrote:

FYI, I have a super hacky support for vs2017 here:
https://github.com/kayru/tundra/commits/vs2017
Though I really have no idea what I'm doing or how to do it properly :)

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/deplinenoise/tundra/issues/289#issuecomment-328333786>,
or mute the thread

**Comment by kayru** [comment]
I only have vs2015 & 2017 installed. They appear to both work. I'll file a PR, but it would be good if someone could test other VS versions.

**Comment by leidegre** [comment]
LGTM

I used to have all versions of VS installed dating back from 2008 and forward but since I no longer have to write software that supports Windows XP I don't.

I think, that the people that need VS2008 support can use an older version of tundra, if we end up breaking something (because I have used them extensively in the past and they have worked well). I only use VS for debugging nowadays. VS Code has by large replaced my editing needs. Not sure how big of a deal the IDE support will be in the future. More and more IDEs are moving to a file system based approach where you don't need to track project files in a separate file. Though I guess this depends on how your environment is set up to include/exclude files through tundra. _Just my two cents._
